gossipd/seeker.c

```diff
@@ -959,6 +967,40 @@ static void seeker_check(struct seeker *seeker)
 	begin_check_timer(seeker);
 }
 
+/* Mutual recursion */
+static void begin_sync_timer(struct seeker *seeker);
+
+/* Periodically ask for a full sync from a random peer to backfill anything
+ * we might have missed. */
+static void full_sync_random_peer(struct seeker *seeker)
+{
+	/* Select random peer */
+	struct peer *random_peer;
+	struct peer_node_id_map_iter it;
+	random_peer = first_random_peer(seeker->daemon, &it);
+	if (!random_peer) {
+		begin_sync_timer(seeker);
+		return;
+	}
+	/* FIXME: store random sync peer and don't select next time. */
+	status_peer_debug(&random_peer->id,
+			  "seeker: chosen for periodic full sync");
+	normal_gossip_start(seeker,random_peer, true);
```

Comment:
This doesn't do a sync, it just starts gossip from this point.

You need to send query_channel_range, then query_short_channel_ids for any we didn't know about in the reply.

Reply:
The `ask_for_all` flag is set, so we should already get everything without a query.  Are you saying we should just be streaming from current and then performing a channel query to fill in gaps?  I think the benefit of just asking for all is we backfill any channel updates we may have missed (for channels we already know about.)

Reply:
I'm not sure this will work with LND nodes, but it will with CLN nodes...